src/package.rs

```diff
@@ -173,16 +184,84 @@ impl PackageStore {
             );
         }
 
-        let pkg_path = fs::canonicalize(&Self::PROTO_PATH)
-            .await
-            .into_diagnostic()
-            .wrap_err({
-                miette!(
-                    "failed to locate package folder (expected directory {} to be present)",
-                    Self::PROTO_PATH
-                )
-            })?;
+        let pkg_path = self.proto_path();
+        let mut entries = BTreeMap::new();
+
+        for entry in self.collect(&pkg_path).await {
+            let path = entry.strip_prefix(&pkg_path).into_diagnostic()?;
+            let contents = tokio::fs::read(&entry).await.unwrap();
+            entries.insert(path.into(), contents.into());
+        }
+
+        let package = Package::create(manifest, entries)?;
+
+        tracing::info!(
+            ":: packaged {}@{}",
+            package.manifest.package.name,
+            package.manifest.package.version
+        );
+
+        Ok(package)
+    }
+
+    /// Directory for the vendored installation of a package
+    pub fn locate(&self, package: &PackageName) -> PathBuf {
+        self.proto_vendor_path().join(&**package)
+    }
+
+    /// Collect .proto files in a given path whilst excluding vendored ones
+    pub async fn collect(&self, path: &Path) -> Vec<PathBuf> {
+        let vendor_path = self.proto_vendor_path();
+        let mut paths: Vec<_> = WalkDir::new(path)
+            .into_iter()
+            .filter_map(Result::ok)
+            .map(|entry| entry.into_path())
+            .filter(|path| !path.starts_with(&vendor_path))
+            .filter(|path| {
+                let ext = path.extension().map(|s| s.to_str());
+
+                matches!(ext, Some(Some("proto")))
+            })
+            .collect();
+
+        paths.sort(); // to ensure determinism
+
+        paths
+    }
+}
+
+#[test]
+fn can_get_proto_path() {
```

Comment:
can you group all tests that you wrote in `#[cfg(test)] mod test` as usual?

Reply:
Sure, I'll do that.